### ISSUE TITLE
C128 VDC support

### ISSUE THREAD

**Issue by xahmol** [EgonOlsen71/petsciiator#2]
Really love the tool. Any chance to have C128 VDC (80 column mode) support? Would require an additional palette for the VDC colors and the option to output in 80x25 instead of 40x25 (preferably with input sized 640 pixel width instead of 320).

**Comment by EgonOlsen71** [comment]
Interesting idea. I haven't done much with the 80 column mode/VDC in the past. I'll look into it when I find the time to see if it's feasible.

**Comment by xahmol** [comment]
Thanks! Browsed your code somewhat, think adding the VDC palette should be little work as it would be just the VIC2Color.java with another 16 color palette RGB values.
Processing from input in 640x200 to 80x25 probably would be more work, but assume that is also doable by parameterisation.

Tried to do something already myself, but I never coded or did anything in Java before, so I could not even compile my first attempt probably by ignorance.

Attached a C128 VDC palette for VICE as reference to get the VDC palette RGB values from.
[c128-vdc-deekay.zip](https://github.com/EgonOlsen71/petsciiator/files/7607173/c128-vdc-deekay.zip)

**Comment by xahmol** [comment]
FYI: I already did use your tool to create this titlescreen: converted the C128D and my picture silhouette to PETSCII with it. Decided to go monochrome and throw away the color output due to the faulty color mapping to VDC palette, but I actually liked the look of the dark grey silhouette. To cope with the 80x25 aspect ratio I doubled the width of the source picture first (keeping the height the same) as in VDC 80 column a character width is half the size than its height in aspect ratio. And did arrange the 40 column data as result to 80 column lines by basically copy pasting hex codes line by line in a hex editor. But would of course way prefer to avoid all those manual steps ;-)



![VDCSE Titlescreen](https://user-images.githubusercontent.com/53264603/143548709-b238c2d2-0c4d-40c0-99c2-8290320bc56c.png)

**Comment by EgonOlsen71** [comment]
Look nice!

**Comment by xahmol** [comment]
Hi, got a message on you closing this.
See completed, does that mean that you do now support VDC? That would be great!
Or does it mean that you decided not to support? (Which is of course also OK, your project)

**Comment by EgonOlsen71** [comment]
No, there's no VDC support for now. I closed the issue for maintanance reasons. Usually, nobody cares about an open issue after several years in which nothing happened. It's like admins do: Just shutdown the old server to see if somebody complains. Since you "complained", I'll reopen it again.

**Comment by xahmol** [comment]
Ok, was not meant as a complaint, just as info request 😉
Anyway, thought it would be that as I saw no commit doing anything for VDC.

Fully up to you of course.
Even tempted to do it myself, but I never ever developed in Java myself.
Think I should be able to manage, code seems easy enough to follow, but already setting up a build chain and getting familiar with it costs more time than I wanted to spend....
Even tried shortly after few months ago, but had issues already with setting up Maven and then no time and prio anymore to solve.

Anyway: would be great if you add VDC support.
If not, this remains to be a very nice tool 😉
